pkg/cmd/util/clientcmd/factory.go

```diff
@@ -53,83 +57,115 @@ func DefaultClientConfig(flags *pflag.FlagSet) clientcmd.ClientConfig {
 // Factory provides common options for OpenShift commands
 type Factory struct {
 	*kubecmd.Factory
-	OpenShiftClientConfig clientcmd.ClientConfig
+	OpenShiftClientConfig ClientConfig
+}
+
+// intercept errors that need special treatment for end-users (e.g. no config file present)
+type ClientConfig struct {
+	KClientConfig clientcmd.ClientConfig
+}
+
+func (cfg *ClientConfig) ClientConfig() (*kclient.Config, error) {
+	c, err := cfg.KClientConfig.ClientConfig()
+	err = DecorateErrors(err)
+	return c, err
+}
+
+func (cfg *ClientConfig) RawConfig() (clientcmdapi.Config, error) {
+	c, err := cfg.KClientConfig.RawConfig()
+	err = DecorateErrors(err)
+	return c, err
+}
+
+func (cfg *ClientConfig) Namespace() (string, error) {
+	n, err := cfg.KClientConfig.Namespace()
+	err = DecorateErrors(err)
+	return n, err
 }
 
 // NewFactory creates an object that holds common methods across all OpenShift commands
 func NewFactory(clientConfig clientcmd.ClientConfig) *Factory {
 	mapper := ShortcutExpander{kubectl.ShortcutExpander{latest.RESTMapper}}
 
-	w := &Factory{kubecmd.NewFactory(clientConfig), clientConfig}
+	w := &Factory{kubecmd.NewFactory(clientConfig), ClientConfig{clientConfig}}
 
 	w.Object = func(cmd *cobra.Command) (meta.RESTMapper, runtime.ObjectTyper) {
 		return mapper, api.Scheme
 	}
 
-	// Save original RESTClient function
-	kRESTClientFunc := w.Factory.RESTClient
 	w.RESTClient = func(cmd *cobra.Command, mapping *meta.RESTMapping) (resource.RESTClient, error) {
+		oClient, kClient, err := w.Clients(cmd)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create client %s: %v", mapping.Kind, err)
+		}
+
 		if latest.OriginKind(mapping.Kind, mapping.APIVersion) {
-			cfg, err := w.OpenShiftClientConfig.ClientConfig()
```

Comment:
Not required, but I wanted to reduce the number of places we call `client.New(cfg)` in favor of `factory.Clients(cmd)`. Can be reverted if not desired.

----- Original Message -----
> >  		if latest.OriginKind(mapping.Kind, mapping.APIVersion) {
> > -			cfg, err := w.OpenShiftClientConfig.ClientConfig()
> 
> Why is this change required?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/992/files#r25139144